sbin/db_mgmt_cpe_dictionary.py

```diff
@@ -19,82 +19,95 @@
 # Imports
 import os
 import sys
+import json
+import hashlib
+import argparse
 runPath = os.path.dirname(os.path.realpath(__file__))
 sys.path.append(os.path.join(runPath, ".."))
 
-from xml.sax import make_parser
-from xml.sax.handler import ContentHandler
-
 from dateutil.parser import parse as parse_datetime
-
-from lib.ProgressBar import progressbar
-from lib.Toolkit import toStringFormattedCPE
 from lib.Config import Configuration
 import lib.DatabaseLayer as db
 
-class CPEHandler(ContentHandler):
-    def __init__(self):
-        self.cpe = []
-        self.titletag = False
-        self.referencestag = False
-        self.referencetag = False
+# parse command line arguments
+argparser = argparse.ArgumentParser(description='populate/update the local CPE database')
+argparser.add_argument('-u', action='store_true', help='update the database')
+argparser.add_argument('-p', action='store_true', help='populate the database')
+argparser.add_argument('-a', action='store_true', default=False, help='force populating the CPE database')
+argparser.add_argument('-f', action='store_true', default=False, help='force update of the CPE database')
+argparser.add_argument('-v', action='store_true', help='verbose output')
+args = argparser.parse_args()
+
+
+def process_cpe_item(item=None):
+    if item is None:
+        return None
+    if "cpe23Uri" not in item:
+        return None
+
+    cpe = {}
+    cpe["title"] = "deprecated"
```

Comment:
New CPE JSON feed does not have this field anymore. What should we do about it?

Reply:
Or maybe to add a human readable version generated from CPE:

`cpe:2.3:h:abb:pcu400:4.4:*:*:*:*:*:*:*` to
`abb pcu400 4.4`

Not sure what's the best.

Reply:
I created a function, generate_title that creates a title based on the CPE string